app/controllers/join_team_requests_controller.rb

```diff
@@ -90,9 +66,31 @@ def destroy
 
   # decline request to join the team...
   def decline
-    @join_team_request = JoinTeamRequest.find(params[:id])
     @join_team_request.status = 'D'
     @join_team_request.save
     redirect_to view_student_teams_path student_id: params[:teams_user_id]
   end
+
+  private
+
+  def check_team
+    # check if the advertisement is from a team member and if so disallow requesting invitations
+    team_member = TeamsUser.where(['team_id =? and user_id =?', params[:team_id], session[:user][:id]])
+    team = Team.find(params[:team_id])
+
+    return flash[:note] = "This team is full." if team.full?
```

Comment:
The logic here is not correct. One can still request, and the request can be saved with a flash message.

See the example below:


class ApplicationController < ActionController::Base
  before_action :require_login
 
  private
 
  def require_login
    unless logged_in?
      flash[:error] = "You must be logged in to access this section"
      redirect_to new_login_url # halts request cycle
    end
  end
end